gateway-service/src/test/java/org/zowe/apiml/acceptance/ZosmfSchemeTest.java

```diff
@@ -9,62 +9,139 @@
+    @Value("${server.ssl.keyAlias:#{null}}")
+    private String keyAlias;
+
+    @Autowired
+    public ZosmfService zosmfService;
 
     @Nested
     class GivenClientCertificate {
         @BeforeEach
         void setUp() throws Exception {
             SslContextConfigurer configurer = new SslContextConfigurer(keystorePassword, clientKeystore, keystore);
             SslContext.prepareSslAuthentication(configurer);
-
             applicationRegistry.clearApplications();
+
             MetadataBuilder defaultBuilder = MetadataBuilder.defaultInstance();
             defaultBuilder.withZosmf();
             applicationRegistry.addApplication(serviceWithDefaultConfiguration, defaultBuilder, false);
             applicationRegistry.setCurrentApplication(serviceWithDefaultConfiguration.getId());
-
             reset(mockClient);
         }
 
         @Nested
         class WhenClientAuthInExtendedKeyUsage {
-            // TODO: add checks for transformation once X509 -> zosmf is implemented
-            @Test
-            void thenOk() throws IOException {
-                mockValid200HttpResponse();
-                given()
-                    .config(SslContext.clientCertUser)
-                    .when()
-                    .get(basePath + serviceWithDefaultConfiguration.getPath())
-                    .then()
-                    .statusCode(is(HttpStatus.SC_OK));
+
+            @Nested
+            class WhenZosmfAuthenticateResponseLTPA {
+
+                @BeforeEach
+                void setZosmfResponse() {
+                    Map<ZosmfService.TokenType, String> tokens = new HashMap<>();
+                    HttpsConfig config = HttpsConfig.builder().keyAlias(keyAlias).keyPassword(keystorePassword).keyStore(keystore).build();
+                    String jwt = JWTUtils.createZoweJwtToken("user", "zosmf", "LTPA_token_from_zosmf", config);
+                    tokens.put(ZosmfService.TokenType.JWT, jwt);
+                    ZosmfService.AuthenticationResponse response = new ZosmfService.AuthenticationResponse("zosmf", tokens);
+                    when(zosmfService.authenticate(any())).thenReturn(response);
+                }
+
+                @Test
+                void thenOk() throws IOException {
+
+                    mockValid200HttpResponse();
+                    given()
+                        .config(SslContext.clientCertUser)
+                        .when()
+                        .get(basePath + serviceWithDefaultConfiguration.getPath())
+                        .then()
+                        .statusCode(is(HttpStatus.SC_OK));
+
+                    ArgumentCaptor<HttpUriRequest> captor = ArgumentCaptor.forClass(HttpUriRequest.class);
+                    verify(mockClient, times(1)).execute(captor.capture());
+
+                    HttpUriRequest toVerify = captor.getValue();
+                    Header cookie = toVerify.getFirstHeader("cookie");
```

Comment:
Could we extract the previous part of the method as a method and us it also below?

Reply:
done

Reply:
thanks